### PR TITLE
[SOAR-9451] Azure Ad Admin update plugin to runtime 4

### PR DESCRIPTION
**Pull request by llaszuk-r7**
### Description

Describe the proposed changes:

  - Update plugin to use InsightConnect Runtime 4

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [ ] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [ ] Screenshot of job output with the plugin changes
- [ ] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [ ] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [ ] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [ ] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [ ] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [ ] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [ ] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [ ] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [ ] Work fully completed
- [ ] Functional
  - [ ] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [ ] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [ ] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [ ] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [ ] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `make validate` and make sure everything passes
2. Run the assessment tool: `icon-plugin run -A -R all -T all`. For single action validation: `icon-plugin run -A -R tests/my_action.json -T tests/my_action.json`
3. Copy (`icon-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/azure_ad_admin/icon_azure_ad_admin/actions/update_user_info/schema.py [hunk 1]
```diff
@@ -48,7 +48,7 @@ class UpdateUserInfoInput(komand.Input):
     "job_title": {
       "type": "string",
       "title": "Job Title",
-      "description": "The user’s job title",
+      "description": "The user is job title",
```
Comment (igorski-r7) [inline plugins/azure_ad_admin/icon_azure_ad_admin/actions/update_user_info/schema.py:51]: Shouldn't that be same as before?
```suggestion
      "description": "The user's job title",
```

**Review by igorski-r7 — CHANGES_REQUESTED** [review]
Looks good! 😃 One single comment left

**Review by igorski-r7 — APPROVED** [review]

File: plugins/azure_ad_admin/Dockerfile [hunk 2]
```diff
@@ -21,8 +21,7 @@ RUN if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
 RUN python setup.py build && python setup.py install
 
 
-
 # Root - Needs SSL CA
```
Comment (mberezin-r7) [inline plugins/azure_ad_admin/Dockerfile:24]:
```suggestion
```
No longer relevant comment if downgrading to `nobody` user.

File: plugins/azure_ad_admin/plugin.spec.yaml [hunk 3]
```diff
@@ -4,9 +4,10 @@ products: [insightconnect]
 name: azure_ad_admin
 title: Azure AD Admin
 description: Perform administrative operations in Azure AD
-version: 2.2.5
+version: 2.2.6
 vendor: rapid7
 support: community
+supported_versions: [2021-01-28]
```
Comment (mberezin-r7) [inline plugins/azure_ad_admin/plugin.spec.yaml:10]: This appears a bit old. Is it correct?

File: plugins/azure_ad_admin/plugin.spec.yaml [hunk 4]
```diff
@@ -418,13 +433,14 @@ actions:
         required: true
   get_group_by_name:
     title: Get Group by Name
-    description: Get a group by it's name
+    description: Get a group by it is name
```
Comment (mberezin-r7) [inline plugins/azure_ad_admin/plugin.spec.yaml:436]:
```suggestion
    description: Get a group by its name
```

File: plugins/azure_ad_admin/plugin.spec.yaml [hunk 5]
```diff
@@ -437,14 +453,16 @@ actions:
...
         title: User ID
-        description: User ID e.g. user@example.com
+        description: User ID
         type: string
         required: true
+        example: user@example.com
       group_name:
         title: Group Name
-        description: Group Name e.g. My Azure Group
+        description: Name of the group to add user
```
Comment (mberezin-r7) [inline plugins/azure_ad_admin/plugin.spec.yaml:462]:
```suggestion
        description: Name of the group to add a user to
```

File: plugins/azure_ad_admin/plugin.spec.yaml [hunk 6]
```diff
@@ -518,7 +538,7 @@ actions:
       job_title:
         title: Job Title
         type: string
-        description: The user’s job title
+        description: The user's job title
```
Comment (mberezin-r7) [inline plugins/azure_ad_admin/plugin.spec.yaml:541]: Wow 😮  good catch!

File: plugins/azure_ad_admin/icon_azure_ad_admin/actions/update_user_info/action.py [hunk 7]
```diff
@@ -29,7 +29,7 @@ def run(self, params={}):
             payload["city"] = city
         if country:
             payload["country"] = country
-        if country:
+        if dept:
```
Comment (mberezin-r7) [inline plugins/azure_ad_admin/icon_azure_ad_admin/actions/update_user_info/action.py:32]: Good catch!!

File: plugins/azure_ad_admin/icon_azure_ad_admin/actions/create_user/action.py [hunk 8]
```diff
@@ -47,10 +48,10 @@ def __init__(self):
     def run(self, params={}):
         passwd = _pw_gen()
 
-        user_created = self._create_user(params, passwd)
+        self._create_user(params, passwd)
 
-        message_to_send = self.compose_email(params, passwd)
-        result = self.send_message(message_to_send, params.get(Input.NOTIFY_FROM))
+        message_to_send = self._compose_email(params, passwd)
```
Comment (mberezin-r7) [inline plugins/azure_ad_admin/icon_azure_ad_admin/actions/create_user/action.py:53]: Wow. How did this work before?? Good catch!

File: plugins/azure_ad_admin/help.md [hunk 9]
```diff
@@ -552,6 +635,7 @@ _This plugin does not contain any troubleshooting information._
 
 # Version History
 
+* 2.2.6 - Update plugin to use InsightConnect Runtime 4
```
Comment (mberezin-r7) [inline plugins/azure_ad_admin/help.md:638]:
```suggestion
* 2.2.6 - Update plugin to use the latest SDK
```

File: plugins/azure_ad_admin/icon_azure_ad_admin/actions/create_user/action.py (hunk 8 above)
Comment (llaszuk-r7) [reply plugins/azure_ad_admin/icon_azure_ad_admin/actions/create_user/action.py:53]: It shouldn't works

**Review by mberezin-r7 — APPROVED** [review]

**Pull request merged**